src/tools/rbd_mirror/ImageReplayer.cc

```diff
@@ -500,6 +510,26 @@ void ImageReplayer<I>::handle_init_remote_journaler(int r) {
     return;
   }
 
+  m_remote_journaler->add_listener(&m_remote_listener);
+
+  cls::journal::Client client;
+  r = m_remote_journaler->get_cached_client(m_local_mirror_uuid, &client);
+  if (r < 0) {
+    derr << "error retrieving remote journal client: " << cpp_strerror(r)
+	 << dendl;
+    on_start_fail(r, "error retrieving remote journal client");
+    return;
+  }
+
+  dout(20) << "client=" << client << dendl;
+
+  if (client.state != cls::journal::CLIENT_STATE_CONNECTED) {
```

Comment:
I would put this logic into the bootstrap state machine since it already pulls the remote peer record -- and eventually it will support auto-resyncing.